lib/benchee/statistics.ex

```diff
@@ -224,35 +224,60 @@ defmodule Benchee.Statistics do
     %__MODULE__{sample_size: 0}
   end
 
-  def job_statistics(run_times) do
-    total_time = Enum.sum(run_times)
-    iterations = Enum.count(run_times)
-    average = total_time / iterations
-    ips = iterations_per_second(average)
-    deviation = standard_deviation(run_times, average, iterations)
-    standard_dev_ratio = deviation / average
-    standard_dev_ips = ips * standard_dev_ratio
-    percentiles = Percentile.percentiles(run_times, [50, 99])
+  def job_statistics(measurements) do
+    total = Enum.sum(measurements)
+    num_iterations = length(measurements)
+    average = total / num_iterations
+    deviation = standard_deviation(measurements, average, num_iterations)
+    standard_dev_ratio = if average == 0, do: 0, else: deviation / average
+    percentiles = Percentile.percentiles(measurements, [50, 99])
     median = Map.fetch!(percentiles, 50)
-    mode = Mode.mode(run_times)
-    minimum = Enum.min(run_times)
-    maximum = Enum.max(run_times)
+    mode = Mode.mode(measurements)
+    minimum = Enum.min(measurements)
+    maximum = Enum.max(measurements)
 
     %__MODULE__{
       average: average,
-      ips: ips,
       std_dev: deviation,
       std_dev_ratio: standard_dev_ratio,
-      std_dev_ips: standard_dev_ips,
       median: median,
       percentiles: percentiles,
       mode: mode,
       minimum: minimum,
       maximum: maximum,
-      sample_size: iterations
+      sample_size: num_iterations
+    }
+  end
+
+  defp standard_deviation(_samples, _average, 1), do: 0
+
+  defp standard_deviation(samples, average, sample_size) do
+    total_variance =
+      Enum.reduce(samples, 0, fn sample, total ->
+        total + :math.pow(sample - average, 2)
+      end)
+
+    variance = total_variance / (sample_size - 1)
+    :math.sqrt(variance)
+  end
+
+  defp add_ips(statistics = %__MODULE{sample_size: 0}), do: statistics
+  defp add_ips(statistics) do
+    ips = iterations_per_second(statistics.average)
+    standard_dev_ips = ips * statistics.std_dev_ratio
+
+    %__MODULE__{
+      statistics |
+      ips: ips,
+      std_dev_ips: standard_dev_ips
     }
   end
 
+  defp iterations_per_second(0.0), do: 0.0
```

Comment:
I thought we wanted `nil` for this case where something seemingly has a runtime of 0.0? We figured that would be a bug, right?

Reply:
oh yeah definitely... thinking in my head was sure then it's 0 but you're right it'd technically be infinity. We could try working the string infinity in - but that might be too much hassle.

Good catch!